pkg/cluster/docker_desktop.go

```diff
@@ -403,3 +403,57 @@ func (c DockerDesktopClient) settingsForWrite(settings interface{}) interface{}
 
 	return settings
 }
+
+type clientRequest struct {
+	client  HTTPClient
+	method  string
+	url     string
+	headers map[string]string
+	body    func() (*bytes.Buffer, error)
+}
+
+func status2xx(resp *http.Response) bool {
+	return resp.StatusCode >= 200 && resp.StatusCode <= 204
+}
+
+func (c DockerDesktopClient) tryRequests(label string, requests []clientRequest) (resp *http.Response, err error) {
+	for _, creq := range requests {
+		if resp != nil {
+			resp.Body.Close()
```

Comment:
i found it very difficult to reason about who's responsible for closing the body. maybe the answer is to break out a `tryOneRequest` method that handles it on non-successful requests?

Reply:
Tried this, I think it's better, let me know what you think.